### ISSUE TITLE
Problem with multiple html pages

### ISSUE THREAD

**Issue by keithslater** [wkhtmltopdf/wkhtmltopdf#2257]
I am running into a problem with wkhtmltopdf and using multiple html pages. This is the command I am running:

```
/usr/local/bin/wkhtmltopdf --print-media-type https://www.example.com/contract.php\?estimate_id=3366 http://www.example.com/work_scope.php\?estimate_id=3366 /tmp/output.pdf
```

On both my local machine and on the server I am running version - wkhtmltopdf 0.12.1 (with patched qt)

The command works perfectly when I run it locally. No problems at all. When I run it on the server (CentOS release 5.10), the command seems to run just fine, no errors. However it doesn't return both html pages in the pdf. The pdf actually will sometimes be the first html page and if I run it again it might be the 2nd page. It just randomly goes back and forth. Like I said though, when I run it locally it combines them with no problem.

I did test by adding a 3rd page and that works fine. It will go back and forth on the first page and it will always display the 3rd page. 

I also ran the command on both pages individually and no problems there.

This is the output from the command

Loading pages (1/6)
Counting pages (2/6)
Resolving links (4/6)
Loading headers and footers (5/6)
Printing pages (6/6)
Done

Anyone else have a similar problem to this? Is there anything else I can do test?

**Comment by rumbu** [comment]
Same issue on Windows using stable 0.12.2.2. Not every page is included in resulting pdf randomly.

**Comment by ashkulz** [comment]
Without a minimal, reproducible test case as requested in the [support page](http://wkhtmltopdf.org/support.html) this issue cannot be investigated further.

**Comment by rumbu** [comment]
OS: Windows 7 SP1;
Command: 
```
"C:\Program Files\wkhtmltopdf\bin\wkhtmltopdf.exe" --print-media-type --image-quality 70 --orientation Portrait --page-size A4 --cache-dir cache ^
 "http://bhaurac.com/" ^
 "http://bhaurac.com/index.php?option=com_content&view=article&id=5&Itemid=2" ^
web.pdf
```
Command output:
```
Loading pages (1/6)
Counting pages (2/6)
Resolving links (4/6)
Loading headers and footers (5/6)
Printing pages (6/6)
Done
```
Result pdf: http://puu.sh/hpfUX/3b6a05d5b7.pdf

**Comment by markdomansky** [comment]
giving this a bump, hoping that it's an easy fix and we can see it soon.  I still see this issue in 0.12.3.1 on Windows.

I would role back to a .11 release, but they aren't available and the one I had doesn't work in Windows 2012 R2.

**Comment by xisket** [comment]
I'm using version 0.12.2.1 and I am experiencing the same problem, trying to print 2 web pages into a single pdf failed, it was just inserting one of  the web pages.
My solution was to include the argument **‐‐javascript‐delay 1000**  (200 by default) , it seems that there is an script into the pages that avoided the page to load.

**Comment by BigNerd** [comment]
I am using wkhtmltopdf 0.12.3 (with patched qt) on Mac OS X El Capitan 10.11.6 (15G1611) and also get the same phenomena: want to print 2 web pages specified as files in the local file system and randomly get either just the first or the second HTML in the PDF output.
Here is the command line I have been using for testing:
/usr/local/bin/wkhtmltopdf /var/folders/ks/vmng3tc51nb43xkklk7jjxxh0000gn/T/java-wkhtmltopdf-wrapper624b2543-cc02-407c-9c07-a8dbaf7d2a03733144934446207671.html /var/folders/ks/vmng3tc51nb43xkklk7jjxxh0000gn/T/java-wkhtmltopdf-wrapper98ff7260-193f-4f0b-9ecb-fa6e4e4938c32120429165745765132.html out.pdf

When adding the above suggested argument "--javascript-delay 1000" like so
/usr/local/bin/wkhtmltopdf --javascript-delay 1000 /var/folders/ks/vmng3tc51nb43xkklk7jjxxh0000gn/T/java-wkhtmltopdf-wrapper624b2543-cc02-407c-9c07-a8dbaf7d2a03733144934446207671.html /var/folders/ks/vmng3tc51nb43xkklk7jjxxh0000gn/T/java-wkhtmltopdf-wrapper98ff7260-193f-4f0b-9ecb-fa6e4e4938c32120429165745765132.html  out.pdf
I get the correct result, i. e. a PDF with both HTMLs as content.
This is really weird and the fix even more. Can someone please explain what the problem is and why this fix is helping?

**Comment by Tomsgu** [comment]
Closing as it seems to be fixed. Please reopen the issue or comment here if the issue still persists.
